### PR TITLE
ZP-1684: Update Pipeline to Fix array_reduce() Bug

### PR DESCRIPTION
**Pull request by Mr-Ming**
## This Pr:
* [x] Upgrade league/pipelin to 0.3.0 to fix a bug with array_reduce() that the league/pipeline made.
* [x] Jira Ticket: https://refinery29.atlassian.net/browse/ZP-1684

## Our NewRelic:
https://rpm.newrelic.com/accounts/60268/applications/12511608/traced_errors/b1f4e05b-96ff-11e7-848b-0242ac11000a_831_2807/similar_errors?original_error_id=29074c54-67f4-11e7-a209-0242ac110010_0_2059

stacktrace shows league/pipeline is the issue and a member in that repo fixed the problem, this is a known problem within thephpleague/pipeline github page;

## Extra Note

I follow the upgrade instruction from 

https://github.com/thephpleague/pipeline/blob/2a384f086d46e17b4590f9d091cbc1a5d1495880/UPGRADE.md

Only Travis issue is:

```
Problem 1
    - humbug/humbug 1.0.0-alpha1 requires phpunit/phpunit ~4.5 -> satisfiable by phpunit/phpunit[4.5.0, 4.5.1, 4.5.x-dev, 4.6.0, 4.6.1, 4.6.10, 4.6.2, 4.6.3, 4.6.4, 4.6.5, 4.6.6, 4.6.7, 4.6.8, 4.6.9, 4.6.x-dev, 4.7.0, 4.7.1, 4.7.2, 4.7.3, 4.7.4, 4.7.5, 4.7.6, 4.7.7, 4.7.x-dev, 4.8.0, 4.8.1, 4.8.10, 4.8.11, 4.8.12, 4.8.13, 4.8.14, 4.8.15, 4.8.16, 4.8.17, 4.8.18, 4.8.19, 4.8.2, 4.8.20, 4.8.21, 4.8.22, 4.8.23, 4.8.24, 4.8.25, 4.8.26, 4.8.27, 4.8.28, 4.8.29, 4.8.3, 4.8.30, 4.8.31, 4.8.32, 4.8.33, 4.8.34, 4.8.35, 4.8.36, 4.8.4, 4.8.5, 4.8.6, 4.8.7, 4.8.8, 4.8.9, 4.8.x-dev] but these conflict with your requirements or minimum-stability.
    - Installation request for humbug/humbug ^1.0.0-alpha1 -> satisfiable by humbug/humbug[1.0.0-alpha1, 1.0.0-alpha2].
    - humbug/humbug 1.0.0-alpha2 requires padraic/phpunit-accelerator ^1.0.2 -> satisfiable by padraic/phpunit-accelerator[v1.0.2].
    - padraic/phpunit-accelerator v1.0.2 requires phpunit/phpunit >=3.7 -> satisfiable by phpunit/phpunit[5.7.20, 5.7.21].
    - phpunit/phpunit 5.7.20 requires sebastian/diff ^1.4.3 -> satisfiable by sebastian/diff[1.4.3].
    - phpunit/phpunit 5.7.21 requires sebastian/diff ^1.4.3 -> satisfiable by sebastian/diff[1.4.3].
    - Conclusion: don't install sebastian/diff 1.4.3
```

### CODE REVIEW

**Comment by Mr-Ming** [comment]
@rcatlin i made a new one since i cant reopen the other one, this pr still needs work, so ill let you know when its ready, seem like a whole bunch of test got broken and when i try to fix one, it breaks something else. going to need time to figure it out

File: src/Middleware/HasMiddleware.php [hunk 1]
```diff
@@ -18,5 +18,5 @@
      *
      * @return $this
      */
-    public function addMiddleware(StageInterface $stage);
+    public function addMiddleware($stage);
```
Comment (Mr-Ming) [inline src/Middleware/HasMiddleware.php]: this is necessary bc our test is passing in callbacks that throws errors

screen shot 2017-09-11 at 3 19 50 pm

File: src/Middleware/HasMiddlewareTrait.php [hunk 2]
```diff
@@ -29,7 +29,7 @@
      *
      * @return $this
      */
-    public function addMiddleware(StageInterface $stage)
+    public function addMiddleware($stage)
```
Comment (Mr-Ming) [inline src/Middleware/HasMiddlewareTrait.php]: this is necessary bc our test is passing in callbacks that throws errors

screen shot 2017-09-11 at 3 19 50 pm

File: src/Middleware/HasMiddleware.php (hunk 1 above)
Comment (Mr-Ming) [inline src/Middleware/HasMiddleware.php]: this is necessary bc our test contains callback to trigger errors

screen shot 2017-09-11 at 3 19 50 pm

File: src/Middleware/HasMiddlewareTrait.php (hunk 2 above)
Comment (Mr-Ming) [inline src/Middleware/HasMiddlewareTrait.php]: this is necessary bc our test contains callback to trigger errors

screen shot 2017-09-11 at 3 19 50 pm

**Comment by rcatlin** [comment]
@Mr-Ming Thanks! I'll look into the remaining Travis/Composer issue and then rebase this PR when it's fixed. Appreciate it!

**Comment by Mr-Ming** [comment]
cool, thanks!

**Pull request merged**